### ISSUE TITLE
Failure to reuse an existing machine and long provisioning time

### ISSUE THREAD

**Issue by bilelomrani1** [skypilot-org/skypilot#2271]
Hi, I'm not sure if these two issues are related but since recently I realized that :
1. `sky launch` recreates a new cluster despite an existing one with the same name already existing
2. The provisioning time has significantly increased.

On issue 2., the process hangs at the following step:

```console
Task from YAML spec: tests/skypilot/config.yml
I 07-19 14:16:[32]optimizer.py:636] == Optimizer ==
I 07-19 14:16:32 optimizer.py:647] Target: minimizing cost
I 07-19 14:16:32 optimizer.py:659] Estimated cost: $0.4 / hour
I 07-19 14:16:32 optimizer.py:659] 
I 07-19 14:16:32 optimizer.py:732] Considered resources (1 node):
I 07-19 14:16:32 optimizer.py:781] --------------------------------------------------------------------------------------------
I 07-19 14:16:32 optimizer.py:781]  CLOUD   INSTANCE        vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
I 07-19 14:16:32 optimizer.py:781] --------------------------------------------------------------------------------------------
I 07-19 14:16:32 optimizer.py:781]  GCP     n1-standard-2   2       7.5       T4:1           us-central1   0.44          ✔     
I 07-19 14:16:32 optimizer.py:781] --------------------------------------------------------------------------------------------
I 07-19 14:16:32 optimizer.py:781] 
Running task on cluster gpu-small-testing-vm...
I 07-19 14:16:32 cloud_vm_ray_backend.py:3856] Creating a new cluster: "gpu-small-testing-vm" [1x GCP(n1-standard-2, {'T4': 1})].
I 07-19 14:16:32 cloud_vm_ray_backend.py:3856] Tip: to reuse an existing cluster, specify --cluster (-c). Run `sky status` to see existing clusters.
I 07-19 14:16:36 cloud_vm_ray_backend.py:1[35] To view detailed progress: tail -n100 -f /root/sky_logs/sky-2023-07-19-14-16-13-057103/provision.log
I 07-19 14:16:38 cloud_vm_ray_backend.py:1703] Launching on GCP us-central1 (us-central1-a)
E 07-19 14:32:50 cloud_vm_ray_backend.py:665] Got the following exception, continuing: Exception: wait_ready timeout exceeded.
I 07-19 14:33:29 cloud_vm_ray_backend.py:1703] Launching on GCP us-central1 (us-central1-b)
I 07-19 14:38:28 cloud_vm_ray_backend.py:1516] Successfully provisioned or found existing VM.
```

The process hangs like at `Launching on GCP us-central1` for ~15min. 

<details close>
<summary>The `provision.log`, looks like this</summary>
<br>

```console
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:03,562 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:08,565 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:08,565 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:09,484 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:14,490 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:14,490 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:15,413 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:20,419 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:20,419 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:21,388 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:26,394 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:26,394 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:27,317 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:32,323 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:32,323 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:33,275 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:38,281 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:38,281 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:39,242 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:44,248 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:44,248 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:45,178 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:50,184 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:50,185 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:51,110 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:24:56,116 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:24:56,117 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:24:57,036 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:02,042 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:02,043 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:02,990 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:07,997 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:07,997 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:08,944 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:13,950 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:13,950 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:14,880 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:19,886 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:19,886 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:20,825 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:25,831 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:25,831 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:26,786 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:31,791 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:31,791 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:32,713 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:37,718 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:37,718 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:38,664 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:43,670 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:43,671 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:44,631 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:49,637 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:49,637 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:50,561 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:25:55,567 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:25:55,568 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:25:56,526 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:26:01,532 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:26:01,532 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:26:02,468 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:26:07,474 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:26:07,474 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.136.205.63' (ECDSA) to the list of known hosts.
gcpuser@34.136.205.63: Permission denied (publickey).
2023-07-19 14:26:08,396 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 14:26:13,398 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 14:26:13,398 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.136.205.63 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
```

</details>

Basically it is a repetition of `SSH still not available (SSH command failed.), retrying in 5 seconds`, despite the fact that the VM (visible on the GCP console) is up, and I'm able to SSH into it manually.

The job eventually launches, proceeds as expected and at the end 

```console
W 07-19 14:46:53 backend_utils.py:2185] Detailed reason: Found 2 node(s) with the same cluster name tag in the cloud provider for cluster 'gpu-small-testing-vm', which should have 1 nodes. This normally should not happen. Please check the cloud console and fix any possible resources leakage (e.g., if there are any stopped nodes and they do not have data or are unhealthy, terminate them).
```

I already terminated the duplicated nodes, but each new pipeline recreates a new machine. I'm not sure if the issue comes from GCP or SkyPilot on this one. Thank you for your help or your insight!

**Comment by concretevitamin** [comment]
Thanks for the report @bilelomrani1! We're looking into this.

There are a couple of abnormal things on first glance:
- Waiting for SSH should show `ssh: connect to host 34.70.132.151 port 22: Connection refused` rather than your log's `gcpuser@34.136.205.63: Permission denied (publickey).`. This may suggest a key setup issue.
- Issue 1 is unexpected
  - After `Exception: wait_ready timeout exceeded.` in us-central1-a, before we proceed to launch in another zone -b we should've torn down any node in zone -a first.

A couple of questions:
-  What version/commit are you on?
- If you still have the full provision.log, it'd be useful to see it (feel free to send privately)

**Comment by bilelomrani1** [comment]
Thanks for your quick response @concretevitamin! This is using the latest Skypilot docker image, I will try to reproduce on the latest commit later today. You're right about the key setup, the weird thing is that this seems to be a new issue, I will double check if our permission set has changed recently. I am re-running a pipeline, I will send you the full `provision.log` file

**Comment by bilelomrani1** [comment]
I managed to trace it down to the docker image specifically: I reproduced our CI setup locally using the exact same service account key. When launching the `sky launch` from my local setup, everything works as expected. I then mounted the service account key into a docker container running on `berkeleyskypilot/skypilot:latest` and I then get the repeated 
```console
2023-07-19 18:24:30,064 INFO updater.py:312 -- SSH still not available (SSH command failed.), retrying in 5 seconds.
2023-07-19 18:24:35,070 VINFO command_runner.py:371 -- Running `uptime`
2023-07-19 18:24:35,071 VVINFO command_runner.py:373 -- Full command is `ssh -tt -i ~/.ssh/sky-key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ExitOnForwardFailure=yes -o ServerAliveInterval=5 -o ServerAliveCountMax=3 -o ControlMaster=auto -o ControlPath=/tmp/ray_ssh_63a9f0ea7b/3f1b682b0f/%C -o ControlPersist=10s -o ConnectTimeout=10s gcpuser@34.170.8.15 bash --login -c -i 'source ~/.bashrc; export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (uptime)'`
Warning: Permanently added '34.170.8.15' (ECDSA) to the list of known hosts.
gcpuser@34.170.8.15: Permission denied (publickey).
```

I will send you the full `provision.log` once the job finishes.

The issue with the clustering re-creation seems also to come from the Docker image specifically, I cannot reproduce it outside of the container.

**Comment by Michaelvll** [comment]
This issue has been resolved offline by mounting the necessary folders to the docker container:
```
-v "$HOME/.config/gcloud:/root/.config/gcloud" \
-v "$HOME/.ssh/sky-key.pub:/root/.ssh/sky-key.pub" \
-v "$HOME/.ssh/sky-key:/root/.ssh/sky-key" \
-v "$HOME/.sky:/root/.sky"
```

This is assuming the host where the containers are launched was the original launcher of the VM.
In which case, we mount the original launcher’s pub/private key, to log in, and the `.sky` directory, to make sure the same VM can be referred to by name inside the container.

If we don’t have the original {pub key, private key, .sky}: The best workaround we can come up with is that we need to terminate the existing VM manually, then redo sky launch from host, so that the host that launches the containers will have these 3 files/dirs, to later mount into the containers

(The reason it worked before and these issues suddenly showed up is due to [#2119](https://github.com/skypilot-org/skypilot/pull/2119).)

The longer provisioning time should be solved by #2273. Closing this issue for now. Please feel free to file another one if there is any unsolved problem.
